core/base/ftmTreePP/FTMTreePP.h

```diff
@@ -85,14 +102,20 @@ namespace ttk {
         const SimplexId mp) {
         AtomicUF *uf = nodesUF_[current].find();
         const SimplexId curVert = tree->getNode(current)->getVertexId();
-        const scalarType curVal = getValue<scalarType>(curVert);
+        const scalarType curVal = (not useCustomTree)
+                                    ? getValue<scalarType>(curVert)
+                                    : tree->getValue<scalarType>(curVert);
 
         for(const auto nodeid : uf->getOpenedArcs()) {
           const SimplexId tmpVert = nodesUF_[nodeid].find()->getExtrema();
           AtomicUF::makeUnion(uf, &nodesUF_[nodeid]);
           if(tmpVert != mp) {
-            const scalarType tmpVal = getValue<scalarType>(tmpVert);
-            if(scalars_->isLower(tmpVert, curVert)) {
+            const scalarType tmpVal = (not useCustomTree)
+                                        ? getValue<scalarType>(tmpVert)
+                                        : tree->getValue<scalarType>(tmpVert);
+            if((useCustomTree and tmpVal < curVal)
+               or (not useCustomTree and scalars_->isLower(tmpVert, curVert))) {
+              // if(scalars_->isLower(tmpVert, curVert)) {
```

Comment:
you may remove old stuff :) git is here if we need them

Reply:
Indeed! Done in the next commit!